File: Assignment_45/question4.java
//////////////////////////////////////////////////////////////////////////////////////////////
//
//  
//  File Name     : question4.java
//  Description   : Check whether matrix is an identity matrix
//  Problem       : Identity matrix: 1s on diagonal, 0s elsewhere
//
//  Author        : Asawari Jadhav
//
//  Date          : 30-07-2025
// 
//////////////////////////////////////////////////////////////////////////////////////////////


class MatrixX
{
    public boolean ChkIdentity(int Arr[][], int iRow, int iCol)
    {
        if(iRow != iCol)
        {
             return false;
        }
           

        for(int i = 0; i < iRow; i++)
        {
            for(int j = 0; j < iCol; j++)
            {
                if(i == j && Arr[i][j] != 1)
                {
                    return false;
                }
                    
                else if(i != j && Arr[i][j] != 0)
                {
                    return false;
                }
                    
            }
        }
        return true;
    }
}

public class question4
{
    public static void main(String args[])
    {
        int[][] Arr = {
            {1, 0, 0},
            {0, 1, 0},
            {0, 0, 1}
        };

        MatrixX mobj = new MatrixX();
        boolean bRet = mobj.ChkIdentity(Arr, 3, 3);

        if(bRet)
        {
            System.out.println("Matrix is Identity Matrix");
        }
        else
        {
            System.out.println("Matrix is not Identity Matrix");
        }
            
    }
}

File: Assignment_24/question4.c
///////////////////////////////////////////////////////////////////////////////////
//
//  File Name   : question4.c
//  Description : Check whether string contains vowels or not.
//  Problem     : Write a program which accepts string from user and checks whether
//                it contains vowels in it or not.
//
//                Input  : "marvellous" → TRUE
//                Input  : "Demo" → TRUE
//                Input  : "xyz" → FALSE
//
//  Author      : Asawari Jadhav
//  Date        : 30/05/2025
//
///////////////////////////////////////////////////////////////////////////////////
#include <stdio.h>

#define TRUE 1
#define FALSE 0

typedef int BOOL;

BOOL ChkVowel(char *str)
{
  while (*str != '\0')
  {
    if (*str == 'A' || *str == 'E' || *str == 'I' || *str == 'O' || *str == 'U' ||
        *str == 'a' || *str == 'e' || *str == 'i' || *str == 'o' || *str == 'u')
    {
      return TRUE;
    }
    str++;
  }
  return FALSE;
}

int main()
{
  char arr[20];
  BOOL bRet = FALSE;

  printf("Enter string: ");
  scanf("%[^'\n']s", arr);

  bRet = ChkVowel(arr);

  if (bRet == TRUE)
  {
    printf("Contains Vowel\n");
  }
  else
  {
    printf("There is no Vowel\n");
  }

  return 0;
}
